### PR TITLE
feat: Issue Credential - functions were exposed for the async usage

### PR DESCRIPTION
**Pull request by soluchok**
The following functions were added:
* `ActionContinue` allows to proceed working with the protocol
* `ActionStop` allows to stop the protocol
* `Actions` returns all available actions

Part of #1173 
Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/issuecredential/service.go [hunk 1]
```diff
@@ -63,14 +67,40 @@ type metaData struct {
...
+type Action struct {
+	// protocol state machine identifier
+	PIID string
+	Msg  service.DIDCommMsgMap
+}
+
+// Opt describes option signature for the Continue function
+type Opt func(md *metaData)
+
+// WithProposeCredential allows to provide ProposeCredential message
```
Comment (DRK3) [inline pkg/didcomm/protocol/issuecredential/service.go]: (optional/low priority)

"allows to provide" sounds odd to me... consider "allows providing" (similar to how you used "allows proceeding" in another area in AllowContinue method) or "allows for the ability to provide a ProposeCredential messsage"? (also for the other "allows to provide" messages in this PR)

File: pkg/didcomm/protocol/issuecredential/states_test.go [hunk 2]
```diff
@@ -402,18 +411,27 @@ func TestProposalSent_CanTransitionTo(t *testing.T) {
...
-	messenger := serviceMocks.NewMockMessenger(ctrl)
-	messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any())
+		messenger := serviceMocks.NewMockMessenger(ctrl)
+		messenger.EXPECT().ReplyTo(gomock.Any(), gomock.Any())
 
-	require.NoError(t, action(messenger))
+		require.NoError(t, action(messenger))
+	})
+
+	t.Run("Error", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (soluchok) [reply pkg/didcomm/protocol/issuecredential/states_test.go]: Done!

**Comment by llorllale** [comment]
I'm having trouble understanding the separation of `service.DIDCommAction` from this new `Action*` concept. Can you explain?

**Comment by soluchok** [comment]
> I'm having trouble understanding the separation of `service.DIDCommAction` from this new `Action*` concept. Can you explain?

Sure. This is for the REST API. For example, your agent received an event. In our service, we are receiving it through the channel. And we can execute to `Continue` or `Stop` function.

What if service crashes or how can we listen to our go specific channel for the REST usage?.

For that purpose, I have exposed three functions.

`Actions()` - returns all available actions we can continue with or stop the protocol.
`ActionContinue(piID string, opt Opt)` -  is equal to `event.Continue(opt Opt)` 
`ActionStop(piID string, err error) `-  is equal to `event.Stop(err)` 

As you can see we can continue with protocol anytime.

**Pull request merged**